osu.Game/Screens/Play/SoloSpectatorScreen.cs

```diff
@@ -54,6 +55,8 @@ public partial class SoloSpectatorScreen : SpectatorScreen, IPreviewTrackOwner
 
         private readonly APIUser targetUser;
 
+        private BeatmapLeaderboardScoresProvider scoresProvider = null!;
```

Comment:
I added scores provider for gameplay leaderboard while spectating solo gameplay, maybe unnecessary for this PR, if so I can remove that sorry.